sycl/include/sycl/types.hpp

```diff
@@ -905,11 +905,46 @@ template <typename Type, int NumElements> class vec {
   // Implement operator [] in the same way for host and device.
   // TODO: change host side implementation when underlying type for host side
   // will be changed to std::array.
-  const DataT &operator[](int i) const {
+  // NOTE: aliasing the incompatible types of bfloat16 may lead to problems if
+  // aggressively optimized. Specializing with noinline to avoid as workaround.
+
+  template <typename T = DataT>
+  typename std::enable_if<!std::is_same_v<T, sycl::ext::oneapi::bfloat16>,
+                          const DataT &>::type
```

Comment:
```suggestion
  typename std::enable_if_t<!std::is_same_v<T, sycl::ext::oneapi::bfloat16>,
                          const DataT &>
```
https://github.com/intel/llvm/pull/12794#discussion_r1504701644 suggests simplifying enable_if part as well.

Reply:
I could swear I did that.  Apparently not.  May have to cut back my day drinking.